webhook/workspace/handler/warning.go

```diff
@@ -0,0 +1,162 @@
+			warnings.eventPostStop = append(warnings.eventPostStop, devWorkspaceSpec.Events.PostStop...)
+		}
+
+		if len(devWorkspaceSpec.Events.PreStop) > 0 {
+			warnings.eventPreStop = append(warnings.eventPreStop, devWorkspaceSpec.Events.PreStop...)
+		}
+	}
+	return warnings
+}
+
+func unsupportedWarningsPresent(warnings *unsupportedWarnings) bool {
+	return len(warnings.serviceAnnotations) > 0 ||
+		len(warnings.endpointAnnotations) > 0 ||
+		len(warnings.dedicatedPod) > 0 ||
+		len(warnings.imageComponent) > 0 ||
+		len(warnings.customComponent) > 0 ||
+		len(warnings.volumeSize) > 0 ||
+		len(warnings.eventPostStop) > 0 ||
+		len(warnings.eventPreStop) > 0
+}
+
+func formatUnsupportedFeaturesWarning(warnings *unsupportedWarnings) string {
+	var msg []string
+	if len(warnings.serviceAnnotations) > 0 {
+		serviceAnnotationsMsg := "components[].container.annotation.service, used by components: " + strings.Join(warnings.serviceAnnotations, ", ")
+		msg = append(msg, serviceAnnotationsMsg)
+	}
+	if len(warnings.endpointAnnotations) > 0 {
+		endpointAnnotationsMsg := "components[].container.endpoints[].annotations, used by components: " + strings.Join(warnings.endpointAnnotations, ", ")
+		msg = append(msg, endpointAnnotationsMsg)
+	}
+	if len(warnings.dedicatedPod) > 0 {
+		dedicatedPodMsg := "components[].container.dedicatedPod, used by components: " + strings.Join(warnings.dedicatedPod, ", ")
+		msg = append(msg, dedicatedPodMsg)
+	}
+	if len(warnings.imageComponent) > 0 {
+		imageComponentMsg := "components[].image, used by components: " + strings.Join(warnings.imageComponent, ", ")
+		msg = append(msg, imageComponentMsg)
+	}
+	if len(warnings.customComponent) > 0 {
+		customComponentMsg := "components[].custom, used by components: " + strings.Join(warnings.customComponent, ", ")
+		msg = append(msg, customComponentMsg)
+	}
+	if len(warnings.volumeSize) > 0 {
+		volumeSizeMsg := "components[].volume.size, used by components: " + strings.Join(warnings.volumeSize, ", ")
+		msg = append(msg, volumeSizeMsg)
+	}
+	if len(warnings.eventPostStop) > 0 {
+		eventPostStopMsg := "events.postStop: " + strings.Join(warnings.eventPostStop, ", ")
+		msg = append(msg, eventPostStopMsg)
+	}
+	if len(warnings.eventPreStop) > 0 {
+		eventPreStopMsg := "events.preStop: " + strings.Join(warnings.eventPreStop, ", ")
+		msg = append(msg, eventPreStopMsg)
+	}
+	// TODO: Improve warning message?
+	return fmt.Sprintf("Unsupported Devfile features are present in this workspace. The following features will have no effect: %s", strings.Join(msg, "; "))
+}
+
+// Returns unsupported feature warnings that are present in the new workspace
+// but not present in the old workspace
+func checkForAddedUnsupportedFeatures(oldWksp, newWksp *dwv2.DevWorkspace) *unsupportedWarnings {
+	oldWarnings := checkUnsupportedFeatures(oldWksp.Spec.Template)
+	newWarnings := checkUnsupportedFeatures(newWksp.Spec.Template)
+	additionalWarnings := &unsupportedWarnings{}
```

Comment:
"Additional" suggests supplementary instead of the diff between the two.
```suggestion
	addedWarnings := &unsupportedWarnings{}
```